torch/_dynamo/variables/builder.py

```diff
@@ -467,12 +466,13 @@ def index_source(key):
                 result = DefaultDictVariable(
                     result,
                     type(value),
-                    self._wrap(value.default_factory),
+                    default_factory=self._wrap(value.default_factory),
+                    source=self.source,
                 )
             else:
-                result = ConstDictVariable(result, type(value))
+                result = ConstDictVariable(result, type(value), source=self.source)
```

Comment:
Is adding all the `source=` in this file needed (throughout the file)?  I thought source was set automatically set for VariableBuilder.

Reply:
Yeah you're right, I think I was considering removing that source assignment at some point when there were more tests failing. I'll remove.

Reply:
Actually, it's needed now for vars where we are going to track them in side effects. I assert that all vars you pass to it have a source which makes sense. This assert wasn't there previously.

Reply:
Perhaps add a helper method on `VariableBuilder` that sets the source and track side effects.